ui/src/app/components/object-detail/object-detail.component.ts

```diff
@@ -548,6 +633,39 @@ export class ObjectDetailComponent implements OnInit {
         },
         maxWidth: '500px',
       })
+    } else if (
+      this.checkIfCkColumn(colId) &&
+      (this.checkIfPkColumn(colId) || associatedIndexes.length != 0)
+    ) {
+      let pkWarning: string = ''
+      let indexWaring: string = ''
+      let connectingString: string = ''
+      let commnaString: string = ''
+      if (this.checkIfPkColumn(colId)) {
+        pkWarning = ` Primary key`
+      }
+      if (associatedIndexes.length != 0) {
+        indexWaring = ` Index ${associatedIndexes}`
+      }
+      if (pkWarning != '' && indexWaring != '') {
+        connectingString = ` and`
+        commnaString = ` ,`
+      }
+      this.dialog.open(InfodialogComponent, {
+        data: {
+          message: `Column ${spColName} is a part of${pkWarning}${commnaString}${indexWaring}${connectingString} check constrainst. Remove the dependencies from respective tabs before dropping the Column. `,
```

Comment:
constrainst - constraints
Please perform a spell check on all the sentences/variables you have added.